services/autoscaling/tests/unit/conftest.py

```diff
@@ -40,12 +75,385 @@ def app_environment(
+                )
+
+    return _runner
+
+
+@pytest.fixture(scope="module")
+def mocked_aws_server() -> Iterator[ThreadedMotoServer]:
+    """creates a moto-server that emulates AWS services in place
+    NOTE: Never use a bucket with underscores it fails!!
+    """
+    server = ThreadedMotoServer(ip_address=get_localhost_ip(), port=unused_port())
+    # pylint: disable=protected-access
+    print(f"--> started mock AWS server on {server._ip_address}:{server._port}")
+    print(
+        f"--> Dashboard available on [http://{server._ip_address}:{server._port}/moto-api/]"
+    )
+    server.start()
+    yield server
+    server.stop()
+    print(f"<-- stopped mock AWS server on {server._ip_address}:{server._port}")
+
+
+@pytest.fixture
+def reset_aws_server_state(mocked_aws_server: ThreadedMotoServer) -> Iterator[None]:
+    # NOTE: reset_aws_server_state [http://docs.getmoto.org/en/latest/docs/server_mode.html#reset-api]
+    yield
+    # pylint: disable=protected-access
+    requests.post(
+        f"http://{mocked_aws_server._ip_address}:{mocked_aws_server._port}/moto-api/reset",
+        timeout=10,
+    )
+
+
+@pytest.fixture
+def mocked_aws_server_envs(
+    app_environment: EnvVarsDict,
+    mocked_aws_server: ThreadedMotoServer,
+    reset_aws_server_state: None,
+    monkeypatch: pytest.MonkeyPatch,
+) -> Iterator[EnvVarsDict]:
+    changed_envs = {
+        "EC2_ENDPOINT": f"http://{mocked_aws_server._ip_address}:{mocked_aws_server._port}",  # pylint: disable=protected-access
+        "EC2_ACCESS_KEY_ID": "xxx",
+        "EC2_SECRET_ACCESS_KEY": "xxx",
+    }
+    yield app_environment | setenvs_from_dict(monkeypatch, changed_envs)
+
+
+@pytest.fixture
+def aws_allowed_ec2_instance_type_names(
+    app_environment: EnvVarsDict,
+    monkeypatch: pytest.MonkeyPatch,
+) -> Iterator[EnvVarsDict]:
+    changed_envs = {
+        "EC2_INSTANCES_ALLOWED_TYPES": json.dumps(
+            [
+                "t2.xlarge",
+                "t2.2xlarge",
+                "r5n.4xlarge",
+                "r5n.8xlarge",
+            ]
+        ),
+    }
+    yield app_environment | setenvs_from_dict(monkeypatch, changed_envs)
+
+
+@pytest.fixture
+def aws_vpc_id(
+    mocked_aws_server_envs: None,
+    app_environment: EnvVarsDict,
+    monkeypatch: pytest.MonkeyPatch,
+    ec2_client: EC2Client,
+) -> Iterator[str]:
+    vpc = ec2_client.create_vpc(
+        CidrBlock="10.0.0.0/16",
+    )
+    vpc_id = vpc["Vpc"]["VpcId"]  # type: ignore
+    print(f"--> Created Vpc in AWS with {vpc_id=}")
+    yield vpc_id
+
+    ec2_client.delete_vpc(VpcId=vpc_id)
+    print(f"<-- Deleted Vpc in AWS with {vpc_id=}")
+
+
+@pytest.fixture
+def aws_subnet_id(
+    monkeypatch: pytest.MonkeyPatch,
+    aws_vpc_id: str,
+    ec2_client: EC2Client,
+) -> Iterator[str]:
+    subnet = ec2_client.create_subnet(CidrBlock="10.0.1.0/24", VpcId=aws_vpc_id)
```

Comment:
vide supra, hardcoded config maybe at the top of the file

Reply:
same as answered above